File: src/main/java/com/rafaelfiume/tictactoe/console/ConsoleGameRenderer.java
package com.rafaelfiume.tictactoe.console;

import com.rafaelfiume.tictactoe.Board;
import com.rafaelfiume.tictactoe.TicTacToePrinter;

import static java.lang.System.lineSeparator;

public class ConsoleGameRenderer {

    public String render(Board gameSnapshot) {
        final TicTacToePrinter print = new TicTacToePrinter(gameSnapshot);
        return String.format(
                "%s" + lineSeparator() +
                "%s" + lineSeparator() + lineSeparator() +
                "%s",
                print.gameDescription(), print.board(), print.gameStatus());
    }

}

File: src/main/java/com/rafaelfiume/tictactoe/console/ConsoleGameRenderer.kt
package com.rafaelfiume.tictactoe.console

import com.rafaelfiume.tictactoe.Game
import com.rafaelfiume.tictactoe.GameRenderer
import com.rafaelfiume.tictactoe.console.ConsoleGamePrinterState.printerFor
import java.lang.System.lineSeparator

class ConsoleGameRenderer : GameRenderer {

    override fun show(game: Game): String {
        val printerState = printerFor(game)

        val ui = lineSeparator() +
                "${printerState.gameDescription()}${lineSeparator()}${lineSeparator()}" +
                "${printerState.board()}${lineSeparator()}${lineSeparator()}" +
                printerState.gameStatus()

        print(ui)
        return ui
    }
}